src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -44,23 +55,56 @@ export class MarkdownHover implements IHoverPart {
 	}
 }
 
+export class VerboseMarkdownHover extends MarkdownHover {
+
+	constructor(
+		owner: IEditorHoverParticipant<MarkdownHover>,
+		range: Range,
+		contents: IMarkdownString[],
+		isBeforeContent: boolean,
+		ordinal: number,
+		public readonly provider: HoverProvider | undefined,
+		public readonly verbosityMetadata: HoverVerbosityMetadata | undefined,
+	) {
+		super(owner, range, contents, isBeforeContent, ordinal);
+	}
+}
+
+interface MarkdownFocusMetadata {
+	index: number | undefined;
+	element: HTMLElement | undefined;
+	focusRemains: boolean;
```

Comment:
This is all a bit unclear. What does `index` refer to? I guess it is the index of the focused element?
I'd remove the `| undefined` (and make `_focusMetadata: MarkdownFocusMetadata | undefined`)

Reply:
That's right, the `index` is the index of the focused element. I'll make it clearer and change the type